backend/inspirehep/pidstore/api/base.py

```diff
@@ -142,3 +142,30 @@ def get_pid_from_record_uri(uri):
         except IndexError:
             return None
         return pid_type, pid_value
+
+    @staticmethod
+    def get_pid_type_from_recid(recid):
+        """Returns the ``pid_type`` of the provided ``recid``.
+
+        Args:
+            recid (str): the ``recid``
+        Returns:
+            The ``pid_type`` if found. None otherwise.
+        """
+        return (
+            PersistentIdentifier.query.with_entities(PersistentIdentifier.pid_type)
+            .filter_by(pid_value=recid)
```

Comment:
there's no index on `pid_value`, so this is a full table scan:
```
explain analyze select pid_type from pidstore_pid where pid_value = '4328';
```
yields:
```
Seq Scan on pidstore_pid  (cost=0.00..118257.46 rows=1 width=4) (actual time=627.701..1416.902 rows=1 loops=1)
  Filter: ((pid_value)::text = '4328'::text)
  Rows Removed by Filter: 3163790
Planning time: 6.127 ms
Execution time: 1416.986 ms
```
we should probably add an index on `pid_type` (the current index is on `(pid_type, pid_value)`, so wrong order for this, and it's useful for the `like` queries on CNUMs).

Reply:
Oh, that's interesting. So for that I should create an alembic migration that creates a `uidx_pid_value` unique index on `pid_value`?

Reply:
you should use the proper naming convention (these indexes are old and don't respect it), so `idx_pidstore_pid_pid_value`, and the uniqueness constraint is not needed nor wanted (there might be collisions in theory between `pid_value` with different `pid_type`. It's probably better to create a partial index where `provider=recid` and add that as an extra filter in the query, as we don't need it for the other external pids.